src/lib/operators/table_scan.cpp

```diff
@@ -0,0 +1,200 @@
+          if (_comparator(type_cast<Type>(_search_value), comparison_value)) {
+            pos_list->push_back(RowID{chunk_id, chunk_offset});
+          }
+        } else {
+          // dictionary Segment
+
+          // cast to dictionary segment
+          auto dictionary_segment = std::dynamic_pointer_cast<DictionarySegment<Type>>(segment);
+
+          // get actual value for position list entry
+          auto comparison_value = type_cast<Type>(dictionary_segment->get(chunk_offset));
+
+          // compare to search value
+          if (_comparator(type_cast<Type>(_search_value), comparison_value)) {
+            pos_list->push_back(RowID{chunk_id, chunk_offset});
+          }
+        }
+      });
+    }
+  } else {
+    // no ref segment
+    auto chunk_count = input_table->chunk_count();
+
+    // iterate over chunks
+    for (auto chunk_id = ChunkID{0}; chunk_id < chunk_count; ++chunk_id) {
+      auto& chunk = input_table->get_chunk(chunk_id);
+      std::shared_ptr<BaseSegment> segment = chunk.get_segment(_column_id);
+
+      // Fill position list for different segment types
+      resolve_data_type(col_type, [&](const auto data_type_t) {
+        using Type = typename decltype(data_type_t)::type;
+
+        // cast to value segment to check whether segment is value segment or dictionary segment
+        auto value_segment = std::dynamic_pointer_cast<ValueSegment<Type>>(segment);
+
+        // get corresponding comparator method for ScanType
+        auto _comparator = _get_comparator_method<Type>();
+
+        // assert that column type and search value type are the same
+        Assert((_search_value.type() == typeid(Type)), "Type of search value and column should be the same");
+
+        if (value_segment) {
+          // value Segment
+          auto chunk_size = chunk.size();
+
+          // iterate over rows
+          for (auto chunk_offset = ChunkOffset{0}; chunk_offset < chunk_size; ++chunk_offset) {
+            // get actual value
+            auto comparison_value = type_cast<Type>(value_segment->values()[chunk_offset]);
+
+            // compare and append to position list if true
+            if (_comparator(type_cast<Type>(_search_value), comparison_value)) {
+              pos_list->push_back(RowID{chunk_id, chunk_offset});
+            }
+          }
+        } else {
+          // dictionary Segment
+          auto dictionary_segment = std::dynamic_pointer_cast<DictionarySegment<Type>>(segment);
+          auto chunk_size = chunk.size();
+
+          // iterate over rows
+          for (auto chunk_offset = ChunkOffset{0}; chunk_offset < chunk_size; ++chunk_offset) {
+            // get actual value
+            auto comparison_value = type_cast<Type>(dictionary_segment->get(chunk_offset));
+
+            // compare and append to position list if true
+            if (_comparator(type_cast<Type>(_search_value), comparison_value)) {
+              pos_list->push_back(RowID{chunk_id, chunk_offset});
+            }
```

Comment:
You are not utilizing the Dictionary here. The scan works on actual values (which could be heavy strings) instead of using efficient ValueIDs. This also happens above.